app/models/local-involvement.js

```diff
@@ -144,13 +145,8 @@ export default class LocalInvolvementModel extends AbstractValidationModel {
    */
   async getOtherLocalInvolvements() {
     const worshipAdministrativeUnit = await this.worshipAdministrativeUnit;
-    let relatedLocalInvolvements = await this.store.query('local-involvement', {
-      filter: {
-        'worship-administrative-unit': {
-          id: worshipAdministrativeUnit.get('id'),
-        },
-      },
-    });
+    const relatedLocalInvolvements =
+      await worshipAdministrativeUnit.involvements;
```

Comment:
I'm not sure if this is safe, since relationships are paginated by mu-cl-resources AFAIK. Doesn't this break if there are more than 20 records in that relationship?

What was the issue with the previous code?

Reply:
After seeing your comment, I tried testing again and found a small issue.

If all existing percentages add up to 99% for example, adding a new instance having a 1% value and saving will work correctly. However, assume all existing percentages add up to 98%, you add two instances with each having a value of 1%, saving will lead to an error even though the actual count is equal to 100%.

If I try and print the calculated percentages, I can see `100%` but also `99%`, which I think is because of each new unsaved instance not being aware of the other, and thus not taking their value into account.

Reply:
@Windvis I had not considered that, will have to test this to make sure whether it is a issue.

@Riadabd, good catch. I think your assessment that the new instances are not aware of each other is correct. Will have to take a deeper look to see how to solve this.

I converted this PR to draft to avoid it is accidentally merged before the issues pointed out are resolved.

Reply:
> @Windvis I had not considered that, will have to test this to make sure whether it is a issue.

@mirdono I think it might work if the relationship data was already loaded before (with includes or a similar query as before) which is probably the case since the table is displayed already. I think EmberData returns everything that it has in the store. But if the data is fetched from the API, I think it's paginated.

Looking at it some more, I'm not the biggest fan of the validation setup to be honest :sweat_smile: . If the table contains 10 local involvements, doesn't that mean this logic runs 10 times as well? It feels wrong to hard wire form data and validations to the model classes like this.

Wouldn't it be better to do the "total percentage must be 100%" check somewhere else?

Reply:
@Windvis The local involvements in this route are indeed included when [retrieving the record](https://github.com/lblod/frontend-organization-portal/blob/80b24c585870a4830d2656c45f9c95707568128b/app/routes/organizations/organization/local-involvements/edit.js#L19) in this route. To be sure I manually checked by making 30 entries and the edit page does not paginate, the display page does paginate per 20 entries. (Sidenote, in production the highest number of local involvements for a single worship service is 9.)

You are also correct that the validation is needlessly executed once for each local involvement individually (which all have the same outcome if everything works correctly). Ideally, this validation would be performed only once at the level of the organisation or so. I do not recall exactly why I implemented it the way it is. Possibly to avoid validating the whole organisation, which can result in errors irrelevant to the form at hand, and/or because this way was easier to map error message to their input fields.

Note: this bug is present on PROD at the moment, so I would prefer first fixing the validation in the state that it is. Improving the actual implementation of the validation can be done "later", with the risk that it is postponed indefinitely...